### PR TITLE
20580 styles for alert messages

### PR DESCRIPTION
**Pull request by vcostin**
[`* `#20580` [Styling][Design] Errors and Notifications`](https://community.openproject.org/work_packages/20580)

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://vcostin-feature-20580-errors-and-notifications.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://vcostin-feature-20580-errors-and-notifications.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by floriank** [comment]
Looks good style wise. Could you add an example for a button within the notifications into the LSG?

**Comment by floriank** [comment]
I think there would be need for a modification class that would set the notification box in the top middle of it's parent container.

I am not sure on how to to this in the LSG off the top of my head. Altough there is the possibility of writing JavaScript in the LSG. 

Could you think of a possibility where we already prepare the "floating stack" functionality as displayed in the Epic's visuals? This could be a separate task. If this might be too much, I'll just merge this PR and open another Task.

File: app/assets/stylesheets/content/_notifications.sass [hunk 1]
```diff
@@ -28,33 +28,95 @@
...
+
+
+@mixin messages-icon
+  @include icon-common
+  position: absolute
+  top: rem-calc(11px)
+  left: rem-calc(10px)
+  font-size: rem-calc(18px)
+
 .notification-box
```
Comment (NobodysNightmare) [inline app/assets/stylesheets/content/_notifications.sass:73]: I think the `notification-box` should define the `color` on its own. Currently it is implicitly inherited.

I'd use `$body-font-color` as color here regardless of success/warning/error.

Comment (floriank) [inline app/assets/stylesheets/content/_notifications.sass:73]: ~~I agree. There could be variables for this in the same namespace (`$notification-box--`). I totally fogot for a moment there, that the blue, yellow and green might collide in the MyProject theme - and they might not be accessible, so easy changing should be possible.~~

Edit: I should read the source before commenting on it.

**Comment by NobodysNightmare** [comment]
I am not yet sure if this should or should not be part of this PR, but http://clients.zzmedia.net/openproject/messages/styleguide/ mentions changes to errorneous input fields.

In case those belong into **this** PR:

~~I strongly disagree with the styling w.r.t. the `text-decoration` and the `color`. The color is not accessible in the way it is currently defined. For the sake of a consistent and accessible UI I propose to use `$body-font-color`.~~

~~For purely aesthetical purposes I also propose to drop the `strike-through` text decoration.~~

edit: @floriank told me those don't belong here.

File: app/assets/stylesheets/content/_notifications.sass [hunk 2]
```diff
@@ -28,33 +28,95 @@
...
 // NOTE: these are demo ski
 
+//nm - notification messages
+$nm-font-size: rem-calc(13px)
+$nm-border-radius: rem-calc(2px)
+$nm-box-padding: rem-calc(10px 35px 10px 35px)
+
+$nm-color-error-border: #ca3f3f
+$nm-color-error-icon: #ca3f3f
+$nm-color-error-background: #fedada
```
Comment (NobodysNightmare) [inline app/assets/stylesheets/content/_notifications.sass:39]: I am unsure if we should define those variables (or similar ones) globally.

What we will encounter elsewhere too is the background color (e.g. for errorneous fields) and the border color (also for errorneous fields).

I can also imagine that we will use the same red/yellow/green tones for other kinds of error/success indication throughout the application.

I'd say we should at least provide the most basic colors (red/green as border/background) in the `_variables.sass`, and probably use them here by assigning them to the `$nm-*` variables.

However, note that I might be missing some important concept within our Sass files, so take my proposal with caution, I am not often working on the Sass side ^^

**Comment by floriank** [comment]
> In case those belong into this PR:

Input fields do  not belong to the PR and the the WP.

**Comment by NobodysNightmare** [comment]
I noticed that the LSG does not feature the "simple notes" specified in http://clients.zzmedia.net/openproject/messages/styleguide/

Note that - as for all other message boxes - I propose to use the `$body-font-color` instead of `#444` as I believe the specification is outdated in that regard.

**Comment by floriank** [comment]
@vcostin Could you add the simple notification as well?

**Comment by floriank** [comment]
could you add the `color` property to the `.notification-box` ? I agree with @NobodysNightmare here. for the moment, the `color` could use `$nm-font-color` which is set to the default `$body-font-color`.

I'll guess we will run into some problems with the different themes (dark/myproject) anyway.

**Comment by floriank** [comment]
Quick nagging: Indentation is 2 spaces :wind_chime:

**Comment by vcostin** [comment]
@floriank sass or markdown? in sass I'm using 2 spaces for a tab, in markdown I used the current file indentation.

**Comment by floriank** [comment]
Markdown

**Comment by NobodysNightmare** [comment]
@floriank Tell me if you agree or disagree:

IMO the "simple" case should not be a modifier, but the default. Currently we would never use the `notification-box` without any modifier.

I'd therefore propose to make the "simple" case the default (no modifiers needed) and only have modifiers for `-warning`, `-error` and `-success`.

Note:
This requires a padding to be applied for all modifier classes, instead (like now) one on the base class and one on the `-simple` modifer.
That makes sense because the higher padding is only there to accomodate for the icon on the left. Without the icon, the padding on the left can be smaller (like done for the simple notification).

**Comment by floriank** [comment]
@NobodysNightmare  This actually makes sense and i would agree.

**Comment by vcostin** [comment]
@floriank Should I implement the changes as specified?

**Comment by floriank** [comment]
Yeah, please!

**Comment by NobodysNightmare** [comment]
:warning: Btw. we should merge this into `release/4.3`, shouldn't we?

**Comment by floriank** [comment]
We should - I'll merge it manually, once @vcostin completes the last requirements.

File: app/assets/stylesheets/content/_notifications.sass [hunk 3]
```diff
@@ -28,33 +28,106 @@
 
 // NOTE: these are demo ski
 
+//nm - notification messages
+$nm-font-size: rem-calc(13px)
+$nm-border-radius: rem-calc(2px)
+$nm-box-padding: rem-calc(10px 35px 10px 35px)
+$nm-default-background: #fff
```
Comment (NobodysNightmare) [inline app/assets/stylesheets/content/_notifications.sass]: This variable is not used anywhere...

**Pull request merged**